### PR TITLE
Add median encoder implementation to KBinsDiscretizer

### LINKED ISSUE #8: Implement Median encoder for KBinsDiscretizer

### PR DESCRIPTION
**Pull request by VishwaP98**
Reference Issues/PRs
issue reference: #8
parent PR: #1

What does this implement/fix? Explain your changes.
find the median value in each bin after KBinsDiscretizer allocates each entry to a specific bin.

### CODE REVIEW

**Comment by VishwaP98** [comment]
Screen Shot 2021-03-07 at 12 01 16 AM

File: sklearn/preprocessing/_discretization.py [hunk 1]
```diff
@@ -314,6 +314,10 @@ def transform(self, X):
             encoder =  lambda x: stats.mode(x)[0][0]
             return self._aggregate_encoder_helper(X, Xt, encoder)
 
+        if self.encode == 'median':
+            encoder = lambda x: median(x, axis=0)
```
Comment (linjun9) [inline sklearn/preprocessing/_discretization.py]: please don't assign a lambda function to a value.

**Review by shrutimistry — APPROVED** [review]

File: sklearn/preprocessing/_discretization.py (hunk 1 above)
Comment (VishwaP98) [reply sklearn/preprocessing/_discretization.py]: done

**Comment by VishwaP98** [comment]
Screen Shot 2021-03-07 at 4 19 59 PM

**Review by linjun9 — APPROVED** [review]

**Pull request merged**